# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/CodeGenerator.cpp
# Commit: 538920d2d777
# Full Hash: 538920d2d77751743a51f10cf730f1ce65b5d0ee
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 3d3e3d124767 -r 538920d2d777 js/src/jit/CodeGenerator.cpp
--- a/js/src/jit/CodeGenerator.cpp	Tue Aug 18 00:03:29 2020 +0000
+++ b/js/src/jit/CodeGenerator.cpp	Tue Aug 18 00:04:37 2020 +0000
@@ -14630,7 +14630,7 @@
         MOZ_CRASH("unexpected argument type when calling from ion to wasm");
       case wasm::ValType::Ref:
         switch (sig.args()[i].refTypeKind()) {
-          case wasm::RefType::Any:
+          case wasm::RefType::Extern:
             // AnyRef is boxed on the JS side, so passed as a pointer here.
             argMir = ToMIRType(sig.args()[i]);
             break;
@@ -14700,7 +14700,7 @@
         MOZ_CRASH("unexpected return type when calling from ion to wasm");
       case wasm::ValType::Ref:
         switch (results[0].refTypeKind()) {
-          case wasm::RefType::Any:
+          case wasm::RefType::Extern:
           case wasm::RefType::Func:
             // The wasm stubs layer unboxes anything that needs to be unboxed
             // and leaves it in a Value.  A FuncRef we could in principle leave